components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/util/APIClientTest.java

```diff
@@ -478,7 +490,8 @@ public void testUnexpectedExceptionFromHttpClient() throws Exception {
 
         when(httpClient.execute(any(HttpPost.class))).thenThrow(new ClientProtocolException("Unexpected exception"));
 
-        ActionInvocationResponse apiResponse = apiClient.callAPI("http://example.com", null, "{}");
+        ActionInvocationResponse apiResponse = apiClient.callAPI(ActionType.PRE_ISSUE_ACCESS_TOKEN,
```

Comment:
IMO we need to add a test case to verify the extensibility provided with a mock or some specific class

Reply:
Added a new test for this.